src/main.c

```diff
@@ -262,6 +263,96 @@ static gboolean bundle_start(int argc, char **argv)
 	return TRUE;
 }
 
+static gboolean write_slot_start(int argc, char **argv)
+{
+	GError *ierror = NULL;
+	RaucImage *image = g_new0(RaucImage, 1);
+	RaucSlot *slot = g_new0(RaucSlot, 1);
+	GFileInfo *info = NULL;
+	GInputStream *instream = NULL;
+	GFile *imagefile = NULL;
+	img_to_slot_handler update_handler = NULL;
+
+	g_debug("write_slot_start");
+
+	if (argc < 3) {
+		g_printerr("A target slot name must be provided\n");
+		r_exit_status = 1;
+		goto out;
+	}
+
+	if (argc < 4) {
+		g_printerr("An image must be provided\n");
+		r_exit_status = 1;
+		goto out;
+	}
+
+	if (argc > 4) {
+		g_printerr("Excess argument: %s\n", argv[4]);
+		goto out;
+	}
+
+	/* construct RaucImage with required attributes */
+	imagefile = g_file_new_for_path(argv[3]);
+	instream = (GInputStream*)g_file_read(imagefile, NULL, &ierror);
+
+	if (instream == NULL) {
+		g_printerr("%s\n", ierror->message);
+		g_clear_error(&ierror);
+		r_exit_status = 1;
+		goto out;
+	}
+
+	info = g_file_input_stream_query_info(G_FILE_INPUT_STREAM(instream),
+			G_FILE_ATTRIBUTE_STANDARD_SIZE, NULL, &ierror);
+
+	if (info == NULL) {
+		g_printerr("%s\n", ierror->message);
+		g_clear_error(&ierror);
+		r_exit_status = 1;
+		goto out;
+	}
+
+	image->checksum.size = g_file_info_get_size(info);
+	image->filename = g_strdup(argv[3]);
+
+	/* retrieve RaucSlot */
+	slot = g_hash_table_lookup(r_context()->config->slots, argv[2]);
+
+	if (slot == NULL) {
+		g_printerr("No matching slot found for given slot name\n");
+		r_exit_status = 1;
+		goto out;
+	}
+
+	/* retrieve update handler */
+	update_handler = get_update_handler(image, slot, &ierror);
+	if (update_handler == NULL) {
+		g_printerr("No handler found for image and slot\n");
```

Comment:
`get_update_handler()` already returns a meaningful error message here, thus use it (the more as you already grab it).